src/main/java/com/toptrumps/core/statistics/GameStateCollector.java

```diff
@@ -0,0 +1,85 @@
+package com.toptrumps.core.statistics;
+
+import com.toptrumps.core.player.Player;
+import com.toptrumps.online.api.request.FinalGameState;
+
+import java.util.Map;
+
+public final class GameStateCollector {
+
+    private final int numberOfDraws;
+    private final Player finalWinner;
+    private final int numberOfRounds;
+    private final Map<Player, Integer> roundWinsMap;
+
+    public int getNumberOfDraws() {
+        return numberOfDraws;
+    }
+
+    public Player getFinalWinner() {
+        return finalWinner;
+    }
+
+    public int getNumberOfRounds() {
+        return numberOfRounds;
+    }
+
+    public Map<Player, Integer> getRoundWinsMap() {
+        return roundWinsMap;
+    }
+
+    public GameStateCollector(Builder builder) {
+        this.numberOfDraws = builder.numberOfDraws;
+        this.finalWinner = builder.finalWinner;
+        this.numberOfRounds = builder.numberOfRounds;
+        this.roundWinsMap = builder.roundWinsMap;
+    }
+
+    public static class Builder {
+
+        private int numberOfDraws;
+        private Player finalWinner;
```

Comment:
Does  a player have an associated ID/getID method? Too lazy to check lmao. In the database i've allocated them 1 through 5 (human = 1, AiFive = 5)

Reply:
We already have an ID associated with each player (but it goes from 0 to 4). It would be nice to remain consistent if possible :)